tests/conftest.py

```diff
@@ -33,82 +31,83 @@
 ]
 
 
-def pytest_addoption(parser: _pytest.config.argparsing.Parser) -> None:
-    parser.addoption(
-        "--test_duration",
-        default="short",
-        choices=["short", "long", "all"],
-        help=
-        "Whether to run short tests (the default), long tests, or all tests. A test is considered short if its timeout is "
-        "less than that specified in the config.")
+def pytest_addoption(parser: pytest.Parser) -> None:
+    parser.addoption("--duration",
+                     default="short",
+                     choices=["short", "long", "all"],
+                     help="""Duration of tests, one of short, long, or all.
+                             Tests are short if their timeout < 2 seconds
+                             (configurable threshold). Default: short.""")
+    parser.addoption("--world-size",
+                     default=int(os.environ.get('WORLD_SIZE', 1)),
+                     type=int,
+                     choices=WORLD_SIZE_OPTIONS,
+                     help="""Filters the tests based on their requested world size.
+                             Defaults to 1, and can also be set by the WORLD_SIZE
+                             environment variable. For world_size>1, please launch
+                             with the composer launcher.""")
+
+
+def _get_timeout(item: pytest.Item, default: float):
+    """Returns the timeout of a test, defaults to -1 """
+    _default = pytest.mark.timeout(default).mark
+    timeout = item.get_closest_marker("timeout", default=_default).args[0]
+    return 999 if timeout == 0 else timeout  # timeout(0) means no timeout restrictions
+
+
+def _get_world_size(item: pytest.Item):
+    """Returns the world_size of a test, defaults to 1."""
+    _default = pytest.mark.world_size(1).mark
+    return item.get_closest_marker("world_size", default=_default).args[0]
+
 
+def _validate_world_size(world_size: int):
+    if "WORLD_SIZE" in os.environ and int(os.environ["WORLD_SIZE"]) != world_size:
```

Comment:
Can you add a check that errors if the CLI world size is greater than 1 but the os.environ world size is not set? In the error message, we can refer people to use the composer launch script to run pytest (or the makefile).